src/compiler/parsing/watf/grammar.js

```diff
@@ -514,8 +511,201 @@ export function parse(tokensList: Array<Object>, source: string): Program {
     }
 
     /**
-     * Parses a line into a instruction
+     * Parses an instruction
+     *
+     * WATF:
+     *
+     * instr      :: plaininst
+     *               blockinstr
+     *
+     * blockinstr :: 'block' I:label rt:resulttype (in:instr*) 'end' id?
+     *               'loop' I:label rt:resulttype (in:instr*) 'end' id?
+     *               'if' I:label rt:resulttype (in:instr*) 'else' id? (in2:intr*) 'end' id?
+     *
+     * plaininst  :: 'unreachable'
+     *               'nop'
+     *               'br' l:labelidx
+     *               'br_if' l:labelidx
+     *               'br_table' l*:vec(labelidx) ln:labelidx
+     *               'return'
+     *               'call' x:funcidx
+     *               'call_indirect' x, I:typeuse
+     *
+     * WAST:
+     *
+     * instr:
+     *   <expr>
+     *   <op>
+     *   block <name>? <block_sig> <instr>* end <name>?
+     *   loop <name>? <block_sig> <instr>* end <name>?
+     *   if <name>? <block_sig> <instr>* end <name>?
+     *   if <name>? <block_sig> <instr>* else <name>? <instr>* end <name>?
+     *
+     * expr:
+     *   ( <op> )
+     *   ( <op> <expr>+ )
+     *   ( block <name>? <block_sig> <instr>* )
+     *   ( loop <name>? <block_sig> <instr>* )
+     *   ( if <name>? <block_sig> ( then <instr>* ) ( else <instr>* )? )
+     *   ( if <name>? <block_sig> <expr>+ ( then <instr>* ) ( else <instr>* )? )
+     *
+     * op:
+     *   unreachable
+     *   nop
+     *   br <var>
+     *   br_if <var>
+     *   br_table <var>+
+     *   return
+     *   call <var>
+     *   call_indirect <func_sig>
+     *   drop
+     *   select
+     *   get_local <var>
+     *   set_local <var>
+     *   tee_local <var>
+     *   get_global <var>
+     *   set_global <var>
+     *   <type>.load((8|16|32)_<sign>)? <offset>? <align>?
+     *   <type>.store(8|16|32)? <offset>? <align>?
+     *   current_memory
+     *   grow_memory
+     *   <type>.const <value>
+     *   <type>.<unop>
+     *   <type>.<binop>
+     *   <type>.<testop>
+     *   <type>.<relop>
+     *   <type>.<cvtop>/<type>
      */
+    function parseFuncInstr() {
```

Comment:
This is currently duplicated (with parseInstructionLine), I will remove this when everything is fine